### PR TITLE
Spread routes

### PR DESCRIPTION
**Pull request by ValzM**

### CODE REVIEW

File: app/persistence/spread_repository.py [hunk 1]
```diff
@@ -133,33 +122,37 @@ def partial_update_spread(fertilizer_id, plot_number, new_date, new_spread_quant
...
-    cur.execute(
-        "UPDATE public.spread SET date = COALESCE(%s, date), spread_quantity = COALESCE(%s, spread_quantity) WHERE "
-        "fertilizer_id = %s AND plot_number = %s",
-        (new_date, new_spread_quantity, fertilizer_id, plot_number),
-    )
-    conn.commit()
-    conn.close()
+    try:
+        cur.execute(
+            "UPDATE public.spread SET date = COALESCE(%s, date), spread_quantity = COALESCE(%s, spread_quantity) WHERE "
```
Comment (Eduardo-Cerqueira) [inline app/persistence/spread_repository.py]: remove public redundancy

File: app/persistence/spread_repository.py [hunk 2]
```diff
@@ -133,33 +122,37 @@ def partial_update_spread(fertilizer_id, plot_number, new_date, new_spread_quant
...
-    existing_spread = get_spread_by_fertilizer(fertilizer_id)
-    if not existing_spread:
-        raise Exception("Spread not found for this fertilizer")
     conn = connect_db()
     cur = conn.cursor()
-    cur.execute("DELETE FROM public.spread WHERE fertilizer_id = %s", (fertilizer_id,))
-    conn.commit()
-    conn.close()
+    try:
+        cur.execute("DELETE FROM public.spread WHERE fertilizer_id = %s", (fertilizer_id,))
```
Comment (Eduardo-Cerqueira) [inline app/persistence/spread_repository.py]: remove public redundancy

File: app/routers/spreads.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    :param fertilizer_id: The fertilizer's UUID.
+    :param spread: The new spread data.
+    :return dict: A dict of the query response.
+    """
+    spread_to_update = get_spread_by_fertilizer(fertilizer_id)
+    if not spread_to_update:
+        raise HTTPException(status_code=404, detail="Spread not found")
+    update_spread(
+        fertilizer_id=fertilizer_id,
+        plot_number=spread_to_update[1],
```
Comment (Eduardo-Cerqueira) [inline app/routers/spreads.py]: add spread.plot_number

File: app/routers/spreads.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    :param fertilizer_id: The fertilizer's UUID.
+    :param spread: The new spread data.
+    :return dict: A dict of the query response.
+    """
+    spread_to_update = get_spread_by_fertilizer(fertilizer_id)
+    if not spread_to_update:
+        raise HTTPException(status_code=404, detail="Spread not found")
+    partial_update_spread(
+        fertilizer_id=fertilizer_id,
+        plot_number=spread_to_update[1],
```
Comment (Eduardo-Cerqueira) [inline app/routers/spreads.py]: add spread.plot_number

**Review by Eduardo-Cerqueira — CHANGES_REQUESTED** [review]

File: app/persistence/spread_repository.py (hunk 1 above)
Comment (ValzM) [reply app/persistence/spread_repository.py]: Done

File: app/routers/spreads.py (hunk 3 above)
Comment (ValzM) [reply app/routers/spreads.py]: Done

File: app/routers/spreads.py (hunk 4 above)
Comment (ValzM) [reply app/routers/spreads.py]: Done

**Review by Eduardo-Cerqueira — APPROVED** [review]
Thanks good job !

**Pull request merged**